### PR TITLE
Update SegmentedControl to support images

### PR DESCRIPTION
**Pull request by huwilkes**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes
Added an optional image to the SegmentedItem. If the image is set, we'll only display the image and not the text, but we'll still use the text for accessibility.
Changed pillMaskedLabelsContainerView's name to pillMaskedContentContainerView, since now it holds more than just labels.
Changed the maskedLabels array to be an array of optionals, since I'm also adding an array for the maskedImages, and we don't need the label/image that isn't being used.
Also updated the logic in sizeThatFits, since I was getting weird results where the window's layout guide was nil or something and just returning zero. (the before iPhone portrait picture shows what I'm talking about)
Moved some logic out of for loops, since there were a few places we were doing things every loop when it was never going to change during the loop.

### Verification
| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| SegmentedControl_images_iPad_before | SegmentedControl_images_iPad_after |
| SegmentedControl_images_iPhone_before | SegmentedControl_images_iPhone_after |
| SegmentedControl_images_iPhone_dark_before | SegmentedControl_images_iPhone_dark_after |
| SegmentedControl_images_iPhone_landscape_before | SegmentedControl_images_iPhone_landscape_after |

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/992)

### CODE REVIEW

File: ios/FluentUI/SegmentedControl/SegmentItem.swift [hunk 1]
```diff
@@ -18,8 +21,13 @@ public class SegmentItem: NSObject {
        }
    }
 
-    @objc public init(title: String, isUnread: Bool = false) {
+    @objc public init(title: String,
```
Comment (harrieshin) [inline ios/FluentUI/SegmentedControl/SegmentItem.swift:24]: would we create convenience method with deprecation on the "old" one for objc clients to migrate better?

File: ios/FluentUI/SegmentedControl/SegmentItem.swift [hunk 2]
```diff
@@ -3,11 +3,14 @@
...
 //
 import Foundation
+import UIKit
 
 /// Used for SegmentedControl array of views
 @objc(MSFSegmentItem)
 public class SegmentItem: NSObject {
     public let title: String
+    public let image: UIImage?
+    public let isTemplateImage: Bool
```
Comment (harrieshin) [inline ios/FluentUI/SegmentedControl/SegmentItem.swift:13]: we can't make this optional too?

File: ios/FluentUI/SegmentedControl/SegmentPillButton.swift [hunk 3]
```diff
@@ -35,9 +35,13 @@ class SegmentPillButton: UIButton {
         self.contentEdgeInsets = Constants.insets
 
         let title = item.title
-        self.setTitle(title, for: .normal)
         self.accessibilityLabel = title
         self.largeContentTitle = title
+        if let image = item.image {
+            self.setImage(image.withRenderingMode(item.isTemplateImage ? .alwaysTemplate : .automatic), for: .normal)
```
Comment (harrieshin) [inline ios/FluentUI/SegmentedControl/SegmentPillButton.swift:41]: we can't trust the clients would do the right thing?

Comment (huwilkes) [reply ios/FluentUI/SegmentedControl/SegmentPillButton.swift:41]: Do you mean for the rendering mode? We don't even do the right thing, our placeholder_20 isn't a template image. We do this in a the popup menu, also.

Comment (huwilkes) [reply ios/FluentUI/SegmentedControl/SegmentPillButton.swift:41]: Looks like we do have that image as a template, but if we don't change the rendering mode we just draw the circle in blue.

Comment (huwilkes) [reply ios/FluentUI/SegmentedControl/SegmentPillButton.swift]: Never mind, that must've been before I was setting the tint color, or something. I removed it and it works fine now.

File: ios/FluentUI/SegmentedControl/SegmentItem.swift [hunk 4]
```diff
@@ -18,8 +20,17 @@ public class SegmentItem: NSObject {
...
    }
 
-    @objc public init(title: String, isUnread: Bool = false) {
+    @objc public convenience init(title: String, isUnread: Bool = false) {
+        self.init(title: title,
+                  image: nil,
+                  isUnread: isUnread)
+    }
+
+    @objc public init(title: String,
```
Comment (harrieshin) [inline ios/FluentUI/SegmentedControl/SegmentItem.swift:29]: documentation?

**Comment by harrieshin** [comment]
tested darkmode and accessibility as well?

File: ios/FluentUI/SegmentedControl/SegmentPillButton.swift [hunk 5]
```diff
@@ -35,9 +35,13 @@ class SegmentPillButton: UIButton {
         self.contentEdgeInsets = Constants.insets
 
         let title = item.title
-        self.setTitle(title, for: .normal)
         self.accessibilityLabel = title
         self.largeContentTitle = title
+        if let image = item.image {
+            self.setImage(image, for: .normal)
```
Comment (harrieshin) [inline ios/FluentUI/SegmentedControl/SegmentPillButton.swift:41]: do we need https://developer.apple.com/documentation/uikit/uilargecontentvieweritem/3183932-largecontentimage?language=objc ?

Comment (huwilkes) [reply ios/FluentUI/SegmentedControl/SegmentPillButton.swift:41]: Large content viewer works, shows the image and the title

**Comment by huwilkes** [comment]
> tested darkmode and accessibility as well?

There's an issue with the disabled dark mode colors, but I haven't had a chance to test voice over or keyboard navigation yet.

File: ios/FluentUI/SegmentedControl/SegmentedControl.swift [hunk 6]
```diff
@@ -461,12 +466,12 @@ open class SegmentedControl: UIControl {
...
-            if let windowWidth = window?.safeAreaLayoutGuide.layoutFrame.width {
-                if traitCollection.userInterfaceIdiom == .pad {
-                    maxButtonWidth = max(windowWidth / 2, 375.0)
-                } else {
-                    maxButtonWidth = windowWidth
-                }
+            let windowSafeAreaInsets = window.safeAreaInsets
+            let windowWidth = window.bounds.width - windowSafeAreaInsets.left - windowSafeAreaInsets.right
+            if traitCollection.userInterfaceIdiom == .pad {
+                maxButtonWidth = max(windowWidth / 2, 375.0)
```
Comment (edjamesmsft) [inline ios/FluentUI/SegmentedControl/SegmentedControl.swift:472]: Nit: should there be a comment or a const somewhere explaining why you're potentially padding up to the magic number of 375?

**Review by edjamesmsft — APPROVED** [review]

File: ios/FluentUI/SegmentedControl/SegmentedControl.swift [hunk 7]
```diff
@@ -438,14 +441,16 @@ open class SegmentedControl: UIControl {
     }
 
     open override func sizeThatFits(_ size: CGSize) -> CGSize {
+        guard let window = window else {
+            return CGSize.zero
+        }
+        let screen = window.windowScene?.screen ?? UIScreen.main
```
Comment (sophialee0416) [inline ios/FluentUI/SegmentedControl/SegmentedControl.swift:447]: I could be wrong, but didn't we want to deprecate our use of UIScreen.main?

Comment (huwilkes) [reply ios/FluentUI/SegmentedControl/SegmentedControl.swift:447]: Could be. I don't remember hearing that guidance. But, we're still using it all over the codebase, and I think trying to get the screen from the window, then using UIScreen.main as a fallback is probably alright.

Comment (harrieshin) [reply ios/FluentUI/SegmentedControl/SegmentedControl.swift:448]: yes, uiscreen.main is wrong and we haven't had time to remove it all over the place.

**Review by harrieshin — APPROVED** [review]

**Pull request merged**